### PR TITLE
Generate .editorconfig from current settings

### LINKED ISSUE #5: Generate .editorconfig from current settings
Would be useful with a command that would create a new .editorconfig based on the users current settings.

### PR DESCRIPTION
**Pull request by varemenos**
Fixes #5

### CODE REVIEW

**Comment by varemenos** [comment]
Sorry I sent a PR with unsquashed commits, I went ahead and squashed them

**Comment by jerone** [comment]
Cool :+1:

File: commands/generate.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+'use babel';
+
+var fs = require('fs');
+
+var getProjectRoot = function getProjectRoot() {
+	'use strict';
```
Comment (sindresorhus) [inline commands/generate.js]: Why are you putting this in every function? It's enough to define it on the top of the file, but use strict is moot anyways as you use `use babel`.

File: commands/generate.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
+'use babel';
+
+var fs = require('fs');
+
+var getProjectRoot = function getProjectRoot() {
```
Comment (sindresorhus) [inline commands/generate.js]: Just use anonymous functions.

File: commands/generate.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
+'use babel';
+
+var fs = require('fs');
```
Comment (sindresorhus) [inline commands/generate.js]: You're using `babel`, so take advantage of ES2015, like using `const` instead of `var`.

File: commands/generate.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		output += 'end_of_line = ' + data.end_of_line+ '\n';
+		output += 'insert_final_newline = ' + data.insert_final_newline+ '\n';
+		output += 'trim_trailing_whitespace = ' + data.trim_trailing_whitespace + '\n';
+		output += 'indent_style = ' + data.indent_style+ '\n';
+		if (data.indent_style === 'space') {
+			output += 'indent_size = ' + data.indent_size+ '\n';
+		} else {
+			output += 'tab_width = ' + data.tab_width+ '\n';
+		}
+
```
Comment (sindresorhus) [inline commands/generate.js]: Push to an array instead and just `.join('\n')` at the end.

File: commands/generate.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+		resolve(output);
+	});
+};
+
+var saveConfigFile = function saveConfigFile(output) {
+	'use strict';
+
+	return new Promise (function(resolve, reject) {
+		var projectRoot = getProjectRoot();
+		var files = fs.readdirSync(projectRoot).filter(file => file === '.editorconfig');
```
Comment (sindresorhus) [inline commands/generate.js]: No point in reading the directory. Just try `fs.access` the `.editorconfig` file.

File: commands/generate.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+		resolve();
+	});
+};
+
+module.exports = function() {
+	'use strict';
+
+	var target = 'atom-workspace';
+	var command = 'editorconfig:generate-configuration-file';
+	var callback = function() {
```
Comment (sindresorhus) [inline commands/generate.js]: The callback can be inline.

File: index.js [hunk 7]
```diff
@@ -40,5 +45,7 @@ function init(editor) {
 }
 
 export let activate = () => {
+	'use strict';
```
Comment (sindresorhus) [inline index.js]: Don't change unrelated code.

File: commands/generate.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+	return new Promise(function (resolve) {
+		atom.notifications.addSuccess('.editorconfig file was successfully generated.', {
+			'detail': 'An .editorconfig file was successfully generated based on your current settings.\nIt\'s stored in your project\'s root'
+		});
+
+		resolve();
+	});
+};
+
+module.exports = function() {
```
Comment (sindresorhus) [inline commands/generate.js]: Space after `function`.

File: commands/generate.js [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+	});
+};
+
+var getOutput = function getOutput(data) {
+	'use strict';
+
+	return new Promise(function(resolve) {
+		var output = '# EditorConfig is awesome: http://EditorConfig.org\n';
+		output += '# top-most EditorConfig file\n';
+		output += 'root = ' + data.root+ '\n\n';
```
Comment (sindresorhus) [inline commands/generate.js]: Space before `+`

**Comment by varemenos** [comment]
@sindresorhus thanks for the code review, I will ping you once I'm done fixing the issues of your notes

**Comment by varemenos** [comment]
@sindresorhus ok, I took care of them

File: index.js [hunk 10]
```diff
@@ -1,7 +1,12 @@
 'use babel';
 import editorconfig from 'editorconfig';
+var generateConfig = require('./commands/generate');
 
 function init(editor) {
+	'use strict';
```
Comment (sindresorhus) [inline index.js]: Nope

File: commands/generate.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
+'use babel';
+
+const fs = require('fs');
+
+var getProjectRoot = function () {
```
Comment (sindresorhus) [inline commands/generate.js]: You can change any `var` that doesn't reassign to `const`, otherwise `let`.

File: commands/generate.js [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+		resolve(output);
+	});
+};
+
+var saveConfigFile = function (output) {
+	return new Promise(function(resolve, reject) {
+		var projectRoot = getProjectRoot();
+		try {
+			fs.accessSync(projectRoot + '/.editorconfig', fs.F_OK);
```
Comment (sindresorhus) [inline commands/generate.js]: Use `path.join()`

File: commands/generate.js [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+			reject('.editorconfig file already exists in your project\'s root');
+		} catch (e) {
+			fs.writeFile(projectRoot + '/.editorconfig', output, () => resolve());
```
Comment (sindresorhus) [inline commands/generate.js]: Use `path.join()`

File: commands/generate.js [hunk 14]
```diff
@@ -0,0 +1,101 @@
+'use babel';
+
+const fs = require('fs');
+
+var getProjectRoot = function () {
+	return atom.project.getPaths()[0];
+};
+
+var getConfig = function () {
+	return new Promise(function(resolve) {
```
Comment (sindresorhus) [inline commands/generate.js]: Why are you (ab)using promises for sync operations? All of these functions can just be normal functions returning a value. I don't see any async operations here.

File: commands/generate.js [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+};
+
+var configAlreadyExists = function (e) {
+	atom.notifications.addError(e);
+};
+
+var successfullyGenerated = function (output) {
+	return new Promise(function (resolve) {
+		atom.notifications.addSuccess('.editorconfig file was successfully generated.', {
+			'detail': 'An .editorconfig file was successfully generated based on your current settings.\nIt\'s stored in your project\'s root'
```
Comment (sindresorhus) [inline commands/generate.js]: Needs a `.` at the end of the sentence.

**Comment by varemenos** [comment]
I've updated the patch to adhere to your comments, again sorry for the mess I'm still not well accustomed to ES6.
